appeals/api/src/server/repositories/integrations.repository.js

```diff
@@ -0,0 +1,42 @@
+// @ts-nocheck
+import { databaseConnector } from '#utils/database-connector.js';
+import { mapDefaultCaseFolders } from '#endpoints/documents/documents.mapper.js';
+
+export const createAppeal = async (data) => {
+	const transaction = await databaseConnector.$transaction(async (tx) => {
+		// Create service users
+		// ...
+
+		// Create appeal
+		let appeal = await tx.appeal.create({ data });
+		if (!data.reference) {
+			const ref = createAppealReference(appeal.id);
+			appeal = await tx.appeal.update({
+				where: { id: appeal.id },
+				data: { reference: ref.toString() }
+			});
+		}
+
+		// Create folders
+		await tx.folder.createMany({ data: mapDefaultCaseFolders(appeal.id) });
+
+		// Organise documents
+		// ...
+
+		// Save documents
+		// ...
+
+		return appeal;
+	});
+
+	return transaction;
+};
+
+export const createDocument = async (data) => {
```

Comment:
It doesn't look like this method needs `aync` as it's not doing anything asynchronous

Reply:
it's not doing anything at all at the moment :-)

Reply:
Exactly my point :D as that is currently how it behaves...however if necessary we can leave it on the expectation of future functionality